libcloud/storage/drivers/backblaze_b2.py

```diff
@@ -292,7 +316,7 @@ def download_object_as_stream(self, obj, chunk_size=None):
                                 success_status_code=httplib.OK)
 
     def upload_object(self, file_path, container, object_name, extra=None,
-                      verify_hash=True, headers=None):
+                      verify_hash=True, headers=None, iterator=None):
```

Comment:
Edit: I see you also implemented `upload_object_via_stream` below. Adding `iterator` argument to this method breaks the standard API so one thing we could do is move common functionality into a new "private" method and then utilize this private method in `upload_object` and `upload_object_via_stream`.